### PR TITLE
Update MOM6 templates for coordination with global-workflow updates 

### LINKED ISSUE #1957: Updating MOM_input_template_500 with wave coupling configurations
## Description
In PR#1916, the MOM_input_template_500 is not updated to the one used in the global workflow because the one in GW changes results due to the change for EPBL_LANGMUIR_SCHEME variable when "USE_WAVE" is set to false. Wave group will confirm how to set this variable with "USE_WAVE" setting. A follow up PR will be made to make MOM_input_template_500 consistent in the model and GW.

### PR DESCRIPTION
**Pull request by JessicaMeixner-NOAA**
## PR Author Checklist:

- [x] I have linked PR's from all sub-components involved in section below. 
- [x] I am confirming reviews are completed in ALL sub-component PR's.
- [x] I have run the full RT suite on either Hera/Cheyenne AND have attached the log to this PR below this line:
  - LOG: 
[RegressionTests_hera.log](https://github.com/ufs-community/ufs-weather-model/files/13266824/RegressionTests_hera.log)
- [x] I have added the list of all failed regression tests to "Anticipated changes" section.
- [x] I have filled out all sections of the template.

## Description
Updates to MOM6 templates for coordination with recent changes in global-workflow for https://github.com/NOAA-EMC/global-workflow/pull/1944 and for the 5 deg template file to include wave-coupling variables, update the  EPBL_LANGMUIR_SCHEME  value and update the output to match workflow. 

### Associated UFSWM Issue to close
- Closes https://github.com/ufs-community/ufs-weather-model/issues/1957

### Subcomponent Pull Requests
None 

### Blocking Dependencies
None

### Subcomponents involved:
- [ ] AQM
- [ ] CDEPS
- [ ] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [ ] FV3
- [ ] GOCART
- [ ] HYCOM
- [ ] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [ ] none

### Input data
- [ ] No changes are expected to input data.
- [x] Changes are expected to input data:
  - [x] New input data. 
  - [ ] Updated input data.

The file on hera /scratch1/NCEPDEV/global/glopara/fix/mom6/20220805/500/oceanda_zgrid_25L.nc needs to be added to @[INPUTDATA_ROOT]/MOM6_FIX/500/

### Regression Tests:
- [ ] No changes are expected to any regression test.
- [x] Changes are expected to the following tests:
Due to the change in output and the change of EPBL_LANGMUIR_SCHEME in the 5deg MOM input, the cpld_control_c48_intel will change answers. 

</details>

### Libraries

- [x] Not Needed
- [ ] Needed
  - [ ] Create separate issue in [JCSDA/spack-stack](https://github.com/JCSDA/spack-stack) asking for update to library. Include library name, library version.
  - [ ] Add issue link from JCSDA/spack-stack following this item 

<details><summary>Code Managers Log</summary>

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.
  - [ ] N/A

### Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [x] Gaea
  - [ ] Cheyenne
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- CI
  - [x] Completed
- opnReqTest
  - [ ] N/A
  - [x] Log attached to comment
</details>

### CODE REVIEW

File: tests/tests/cpld_control_c48 [hunk 1]
```diff
@@ -106,6 +106,8 @@ export CHLCLIM=''
 export FRUNOFF=''
 export MOM6_RIVER_RUNOFF=False
 export MOM6_RESTART_SETTING=r
+export MOM6_DIAG_COORD_DEF_Z_FILE=oceanda_zgrid_25L.nc
```
Comment (junwang-noaa) [inline tests/tests/cpld_control_c48:109]: May I ask why the C48 test has different diag coord def file from other resolutions?

File: tests/fv3_conf/cpld_control_run.IN [hunk 2]
```diff
@@ -42,6 +42,7 @@ cp    @[INPUTDATA_ROOT]/${FV3_DIR}/INPUT/grid_spec.nc ./INPUT/@[ATMRES]_mosaic.n
 
 # MOM6 fixed input
 cp    @[INPUTDATA_ROOT]/MOM6_FIX/@[OCNRES]/* ./INPUT
+cp /scratch1/NCEPDEV/climate/Jessica.Meixner/MOM6settings/global-workflow/fix/mom6/500/oceanda_zgrid_25L.nc ./INPUT
```
Comment (junwang-noaa) [inline tests/fv3_conf/cpld_control_run.IN]: I assume this line is not needed once the oceanda_zgrid_25L.nc is copied to input directory.

Comment (JessicaMeixner-NOAA) [reply tests/fv3_conf/cpld_control_run.IN]: Yes, that is correct.

Comment (DeniseWorthen) [reply tests/fv3_conf/cpld_control_run.IN]: I'm not seeing the file in this location: ``/scratch1/NCEPDEV/climate/Jessica.Meixner/MOM6settings/``

Comment (JessicaMeixner-NOAA) [reply tests/fv3_conf/cpld_control_run.IN]: @DeniseWorthen I thought I put in the actual fix file location, not my personal link to the global-workflow fix.  I'll update this now here and on the PR template.

Comment (JessicaMeixner-NOAA) [reply tests/fv3_conf/cpld_control_run.IN]: This is the correct location: /scratch1/NCEPDEV/global/glopara/fix/mom6/20220805/500/oceanda_zgrid_25L.nc

Comment (DeniseWorthen) [reply tests/fv3_conf/cpld_control_run.IN]: Ok, thanks.

File: tests/fv3_conf/cpld_control_run.IN [hunk 3]
```diff
@@ -42,6 +42,7 @@ cp    @[INPUTDATA_ROOT]/${FV3_DIR}/INPUT/grid_spec.nc ./INPUT/@[ATMRES]_mosaic.n
 
 # MOM6 fixed input
 cp    @[INPUTDATA_ROOT]/MOM6_FIX/@[OCNRES]/* ./INPUT
+cp /scratch1/NCEPDEV/global/glopara/fix/mom6/20220805/500/oceanda_zgrid_25L.nc ./INPUT
```
Comment (JessicaMeixner-NOAA) [inline tests/fv3_conf/cpld_control_run.IN]: @jkbk2004 has the input files been put on all the platforms?  If so I can revert this and re-run tests as needed.

Comment (jkbk2004) [reply tests/fv3_conf/cpld_control_run.IN]: to input-data-20221101/MOM6_FIX/500/oceanda_zgrid_25L.nc, right?

Comment (jkbk2004) [reply tests/fv3_conf/cpld_control_run.IN]: rsynced ok input-data-20221101/MOM6_FIX/500/oceanda_zgrid_25L.nc on hera/orion/hercules/gaea/jet. We will clean up cheyenne and replace with Derecho soon.

Comment (JessicaMeixner-NOAA) [reply tests/fv3_conf/cpld_control_run.IN]: > to input-data-20221101/MOM6_FIX/500/oceanda_zgrid_25L.nc, right?

@jkbk2004  that's correct.   I'll remove this line and re-run on hera.

Comment (JessicaMeixner-NOAA) [reply tests/fv3_conf/cpld_control_run.IN]: I did just double check and this will be needed on WCOSS2 as well.

Comment (jkbk2004) [reply tests/fv3_conf/cpld_control_run.IN]: @BrianCurtis-NOAA can you or anyone else follow on wcoss2 side?

**Comment by JessicaMeixner-NOAA** [comment]
Here's the new log file from hera, comparing against a newly created baseline: 
[RegressionTests_hera.log](https://github.com/ufs-community/ufs-weather-model/files/13490707/RegressionTests_hera.log)

File: tests/fv3_conf/cpld_control_run.IN (hunk 3 above)
Comment (JessicaMeixner-NOAA) [reply tests/fv3_conf/cpld_control_run.IN]: It looks like it'll be this PRs turn soon.  I did just double check and we're still missing the new input file input-data-20221101/MOM6_FIX/500/oceanda_zgrid_25L.nc on WCOSS2.

Comment (BrianCurtis-NOAA) [reply tests/fv3_conf/cpld_control_run.IN]: @JessicaMeixner-NOAA: Done.
```
emc.nems@clogin01:/lfs/h2/emc/nems/noscrub/emc.nems/RT/NEMSfv3gfs/input-data-20221101/MOM6_FIX/500> ls
hycom1_25.nc      MOM_override          ocean_hgrid.nc  ocean_mosaic.nc
layer_coord25.nc  oceanda_zgrid_25L.nc  ocean_mask.nc   ocean_topog.nc
```

**Comment by jkbk2004** [comment]
@JessicaMeixner-NOAA Can we delay this pr one or two days more? RRFS operation application requires to merge #2021 ASAP. This week's schedule is not that busy. So, I think we can commit this pr tomorrow or 12/06 right after Dusan's PR.

**Comment by JessicaMeixner-NOAA** [comment]
Please prioritize the work needed for RRFS and thank you for communicating the reasons for the delay.

**Comment by jkbk2004** [comment]
@JessicaMeixner-NOAA can you sync up branch? we can working on this pr.

**Comment by JessicaMeixner-NOAA** [comment]
> @JessicaMeixner-NOAA can you sync up branch? we can working on this pr.

@jkbk2004 this branch has been sync'd.

**Comment by zach1221** [comment]
Testing is complete, let's move to review.

**Review by jkbk2004 — DISMISSED** [review]

**Review by BrianCurtis-NOAA — DISMISSED** [review]

**Comment by zach1221** [comment]
@JessicaMeixner-NOAA looks like there is one outdated conversation. Can you resolve it for us, please?

**Comment by DeniseWorthen** [comment]
I don't understand the hercules log. Why is control_wrtGauss_netcdf_parallel_intel  showing up as test 235? It should be around test #30. It also shows up as a different rt_ directory. Is this a appended log? If so, the original log showing test failures should not be removed. If a second log is appended, it should clearly indicate that it is from a re-run.

Test 235 control_wrtGauss_netcdf_parallel_intel PASS

The test seems to have failed at the initial run, but does not show as a "failure"
https://github.com/ufs-community/ufs-weather-model/blob/aa7dea89bea0f52193653096a78a628ccf244673/tests/logs/RegressionTests_hercules.log#L1976-L1983

The RT seems to simply skip the reporting line? There should be a line like "Test 034 control_wrtGauss_netcdf_parallel_intel FAIL"

**Comment by JessicaMeixner-NOAA** [comment]
> @JessicaMeixner-NOAA looks like there is one outdated conversation. Can you resolve it for us, please?

done

**Comment by zach1221** [comment]
> I don't understand the hercules log. Why is control_wrtGauss_netcdf_parallel_intel showing up as test 235? It should be around test #30. It also shows up as a different rt_ directory. Is this a appended log? If so, the original log showing test failures should not be removed. If a second log is appended, it should clearly indicate that it is from a re-run.
> 
> Test 235 control_wrtGauss_netcdf_parallel_intel PASS
> 
> The test seems to have failed at the initial run, but does not show as a "failure"
> 
> https://github.com/ufs-community/ufs-weather-model/blob/aa7dea89bea0f52193653096a78a628ccf244673/tests/logs/RegressionTests_hercules.log#L1976-L1983
> 
> The RT seems to simply skip the reporting line? There should be a line like "Test 034 control_wrtGauss_netcdf_parallel_intel FAIL"

Yes, it's the nccmp failure on Hercules. I have to run that case a second time to get it to pass. I did clean up the combined logs afterwards, so going forward I won't.

**Comment by DeniseWorthen** [comment]
@zach1221 OK, thanks. I'm also concerned though that there is an actual problem w/ the RT failure detection. I assume you didn't remove the line that reported the original test 034 failed "Test 034 control_wrtGauss_netcdf_parallel_intel FAIL"?

**Comment by zach1221** [comment]
> @zach1221 OK, thanks. I'm also concerned though that there is an actual problem w/ the RT failure detection. I assume you didn't remove the line that reported the original test 034 failed "Test 034 control_wrtGauss_netcdf_parallel_intel FAIL"?

No I didn't remove the actual failed 034 control_wrtGauss_netcdf_parallel_intel results piece, that should still in there.

**Comment by DeniseWorthen** [comment]
@junwang-noaa It seems we do have an issue w/ failure detection. It seems that when we get the "ALT CHECK ERROR" failure, the job is not always correctly reported.

**Comment by zach1221** [comment]
Ok re-pushed the original logs without any edits.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Comment by DeniseWorthen** [comment]
@zach1221 Thanks for posting the original logs. I see that at the end, the test 034 is noted as failed, but at the point where the files are being compared, it doesn't catch the fact that the comparison gives an error. It seems like after this line

Comparing atmf024.nc ............ALT CHECK......ERROR

We should get a report, something like

Test 034 control_wrtGauss_netcdf_parallel_intel FAIL

**Review by jkbk2004 — APPROVED** [review]
I approve as new Hercules log has enough information for the failed case.

**Comment by zach1221** [comment]
> @zach1221 Thanks for posting the original logs. I see that at the end, the test 034 is noted as failed, but at the point where the files are being compared, it doesn't catch the fact that the comparison gives an error. It seems like after this line
> 
> Comparing atmf024.nc ............ALT CHECK......ERROR
> 
> We should get a report, something like
> 
> Test 034 control_wrtGauss_netcdf_parallel_intel FAIL

Ok, yah that's not happening it looks like. I can create a issue for it.

File: tests/tests/cpld_control_c48 (hunk 1 above)
Comment (DeniseWorthen) [inline tests/tests/cpld_control_c48:109]: @JessicaMeixner-NOAA Does this MOM6_DIAG_COORD_DEF_Z_FILE need to be added to the cpld_warmstart_c48 and cpld_restart_c48? Those tests are currently commented out (I need to recreate the warmstart files that they use). Will we need this variable set for those two tests?

Comment (JessicaMeixner-NOAA) [reply tests/tests/cpld_control_c48:109]: @DeniseWorthen good catch, yes you will want to set that for the other C48 tests.

**Pull request merged**